tests/shellcheck/run_shellcheck.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# If $top_srcdir has not been set by automake, import the test environment
+if [ -z "$top_srcdir" ]; then
+    top_srcdir="$(realpath "$(dirname "$0")/../..")"
+    . "${top_srcdir}"/tests/testenv.sh
+fi
+
+source "${top_srcdir}/tests/shellcheck/ignores.sh"
+logfile="${top_srcdir}/tests/shellcheck/shellcheck.log"
+
+# Check if this test can be run
+if ! type shellcheck > /dev/null 2>&1 ; then
+    echo "SKIP - shellcheck must be installed to run it."
+    exit 77
+fi
+
+# Identify which shellcheck version is run for debugging fails due to version bump
+shellcheck --version | tee "$logfile"
+echo | tee -a "$logfile"
+
+# Ignore files according to path patterns
+pushd "${top_srcdir}" > /dev/null || return 1
+files=$(git ls-files)
+# shellcheck disable=SC2154  # comes from the file sourced earlier
+for ignore in "${ignore_patterns[@]}" ; do
+    files=$(echo "$files" | grep -v "$ignore")
+done
+
+# Find files that are shell scripts
+files=$(echo "$files" | xargs file | grep "shell script" | awk -F ":" '{print $1}')
+
+# List files being checked
+echo -e "Files to be checked:\n--------------------\n$files\n" | tee -a "$logfile"
+
+# Do the actual linting
+echo -e "Check results:\n--------------" | tee -a "$logfile"
```

Comment:
The check is not hooked to workflows, right? Could you copy-paste here the current output, please?

Reply:
Sure!

<details>
<summary>Logs</summary>

```
ShellCheck - shell script analysis tool
version: 0.7.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net

Files to be checked:
--------------------
autogen.sh
data/liveinst/liveinst
data/liveinst/liveinst-setup.sh
data/systemd/anaconda-generator
dracut/anaconda-copy-cmdline.sh
dracut/anaconda-copy-dhclient.sh
dracut/anaconda-copy-ks.sh
dracut/anaconda-copy-prefixdevname.sh
dracut/anaconda-copy-s390ccwconf.sh
dracut/anaconda-depmod.sh
dracut/anaconda-diskroot
dracut/anaconda-error-reporting.sh
dracut/anaconda-hmcroot
dracut/anaconda-ifcfg.sh
dracut/anaconda-ifdown
dracut/anaconda-ks-sendheaders.sh
dracut/anaconda-lib.sh
dracut/anaconda-modprobe.sh
dracut/anaconda-netroot.sh
dracut/anaconda-nfsrepo-cleanup.sh
dracut/anaconda-pre-shutdown.sh
dracut/anaconda-set-kernel-hung-timeout.sh
dracut/driver-updates-genrules.sh
dracut/fetch-driver-net.sh
dracut/fetch-kickstart-disk
dracut/fetch-kickstart-net.sh
dracut/fetch-updates-disk
dracut/find-net-intfs-by-driver
dracut/kickstart-genrules.sh
dracut/module-setup.sh
dracut/parse-anaconda-dd.sh
dracut/parse-anaconda-kickstart.sh
dracut/parse-anaconda-net.sh
dracut/parse-anaconda-options.sh
dracut/parse-anaconda-repo.sh
dracut/repo-genrules.sh
dracut/save-initramfs.sh
dracut/updates-genrules.sh
tests/cppcheck/runcppcheck.sh
tests/gettext_tests/canary_tests.sh
tests/glade_tests/glade_tests.sh
tests/lib/testlib.sh
tests/pep8/runpep8.sh
tests/rpm_tests/rpm_tests.sh
tests/shellcheck/ignores.sh
tests/shellcheck/run_shellcheck.sh
tests/testenv.sh
tests/unit_tests/unit_tests.sh

Check results:
--------------
```
</details>

Reply:
Is it hooked to workflows? Indirectly. It runs with unit tests, same as glade tests and the others.

Reply:
Oh, I have missed that, sorry. However, it looks like you don't propagate the exit status, because `shellcheck` found some issues, but the test passed.

```
Check results:
--------------
data/liveinst/liveinst-setup.sh:13:66: note: Not following: ./user-dirs.dirs was not specified as input (see shellcheck -x). [SC1091]
dracut/fetch-driver-net.sh:48:58: note: Double quote to prevent globbing and word splitting. [SC2086]
dracut/parse-anaconda-options.sh:52:49: note: Double quote to prevent globbing and word splitting. [SC2086]
tests/pep8/runpep8.sh:9:3: note: Not following: ./tests/pep8/config.target was not specified as input (see shellcheck -x). [SC1091]
PASS shellcheck/run_shellcheck.sh (exit status: 0)
```

Reply:
Ouch, thanks. I thought it would somehow propagate out from the piping. Back to that, I guess :(